Sources/Commands/SwiftTestTool.swift

```diff
@@ -1139,3 +1140,141 @@ private extension SwiftTool {
         return parameters
     }
 }
+
+extension SwiftTestTool {
+
+    public static func doTest(subset: PluginInvocationTestSubset, parameters: PluginInvocationTestParameters, swiftTool: SwiftTool) throws -> PluginInvocationTestResult {
```

Comment:
should this return something more generic than PluginInvocationTestResult and then the call site does the translation? this feels fairly leaky at the abstraction level

Reply:
Right, that's a good point.  I originally started with this being in SwiftPackageTool but then kept it as it moved over.  Really the plugin callbacks should be in their own separate file, actually, but I was hesitant to "pull on that thread" as part of this PR.  But it's certainly a better place to end up.

I'll move this over as per your previous comment though — that seems better for this PR, and then switching SwiftTestTool to use the same code in a further followup.